### PR TITLE
[Enhancement] Support Amazon VPC Flow Logs to Kinesis Data Firehose

### LINKED ISSUE #26730: VPC Flow Logs can now be delivered to Amazon Kinesis Firehose
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> You can now deliver Amazon Virtual Private Cloud (VPC) Flow Logs directly to Amazon Kinesis Firehose, allowing you to stream your flow logs real-time to [destinations supported by Amazon Kinesis Firehose](https://docs.aws.amazon.com/firehose/latest/dev/create-destination.html) or [downstream logging solutions](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-amazon-vpc-flow-logs-kinesis-data-firehose/) via custom HTTP endpoints.

### New or Affected Resource(s)

* [`aws_flow_log`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/flow_log)

### Potential Terraform Configuration

```hcl
resource "aws_flow_log" "example" {
  log_destination_type = "kinesis-data-firehose"

  deliver_cross_account_role = "..."
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2022/09/amazon-vpc-flow-logs-delivered-amazon-kinesis-firehose/).
[Blog post](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-amazon-vpc-flow-logs-kinesis-data-firehose/).

Requires [AWS SDK for Go v1.44.94](https://github.com/aws/aws-sdk-go/releases/tag/v1.44.94): https://github.com/hashicorp/terraform-provider-aws/pull/26720.

### PR DESCRIPTION
**Pull request by BhanuCh**
### Description

Adds validation for VPC Flow logs destination type as Kinesis Data Firehose

### Relations

Closes #26730

### Output from Acceptance Testing

```
$make testacc TESTARGS='-run=TestAccVPCFlowLog' PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 2  -run=TestAccVPCFlowLog -timeout 180m
=== RUN   TestAccVPCFlowLog_vpcID
=== PAUSE TestAccVPCFlowLog_vpcID
=== RUN   TestAccVPCFlowLog_logFormat
=== PAUSE TestAccVPCFlowLog_logFormat
=== RUN   TestAccVPCFlowLog_subnetID
=== PAUSE TestAccVPCFlowLog_subnetID
=== RUN   TestAccVPCFlowLog_transitGatewayID
=== PAUSE TestAccVPCFlowLog_transitGatewayID
=== RUN   TestAccVPCFlowLog_transitGatewayAttachmentID
=== PAUSE TestAccVPCFlowLog_transitGatewayAttachmentID
=== RUN   TestAccVPCFlowLog_LogDestinationType_cloudWatchLogs
=== PAUSE TestAccVPCFlowLog_LogDestinationType_cloudWatchLogs
=== RUN   TestAccVPCFlowLog_LogDestinationType_kinesisFirehose
=== PAUSE TestAccVPCFlowLog_LogDestinationType_kinesisFirehose
=== RUN   TestAccVPCFlowLog_LogDestinationType_s3
=== PAUSE TestAccVPCFlowLog_LogDestinationType_s3
=== RUN   TestAccVPCFlowLog_LogDestinationTypeS3_invalid
=== PAUSE TestAccVPCFlowLog_LogDestinationTypeS3_invalid
=== RUN   TestAccVPCFlowLog_LogDestinationTypeS3DO_plainText
=== PAUSE TestAccVPCFlowLog_LogDestinationTypeS3DO_plainText
=== RUN   TestAccVPCFlowLog_LogDestinationTypeS3DOPlainText_hiveCompatible
=== PAUSE TestAccVPCFlowLog_LogDestinationTypeS3DOPlainText_hiveCompatible
=== RUN   TestAccVPCFlowLog_LogDestinationTypeS3DO_parquet
=== PAUSE TestAccVPCFlowLog_LogDestinationTypeS3DO_parquet
=== RUN   TestAccVPCFlowLog_LogDestinationTypeS3DOParquet_hiveCompatible
=== PAUSE TestAccVPCFlowLog_LogDestinationTypeS3DOParquet_hiveCompatible
=== RUN   TestAccVPCFlowLog_LogDestinationTypeS3DOParquetHiveCompatible_perHour
=== PAUSE TestAccVPCFlowLog_LogDestinationTypeS3DOParquetHiveCompatible_perHour
=== RUN   TestAccVPCFlowLog_LogDestinationType_maxAggregationInterval
=== PAUSE TestAccVPCFlowLog_LogDestinationType_maxAggregationInterval
=== RUN   TestAccVPCFlowLog_tags
=== PAUSE TestAccVPCFlowLog_tags
=== RUN   TestAccVPCFlowLog_disappears
=== PAUSE TestAccVPCFlowLog_disappears
=== CONT  TestAccVPCFlowLog_vpcID
=== CONT  TestAccVPCFlowLog_LogDestinationTypeS3DO_plainText
--- PASS: TestAccVPCFlowLog_LogDestinationTypeS3DO_plainText (21.89s)
=== CONT  TestAccVPCFlowLog_LogDestinationType_cloudWatchLogs
--- PASS: TestAccVPCFlowLog_vpcID (34.36s)
=== CONT  TestAccVPCFlowLog_LogDestinationTypeS3_invalid
--- PASS: TestAccVPCFlowLog_LogDestinationType_cloudWatchLogs (20.21s)
=== CONT  TestAccVPCFlowLog_transitGatewayID
--- PASS: TestAccVPCFlowLog_LogDestinationTypeS3_invalid (11.94s)
=== CONT  TestAccVPCFlowLog_LogDestinationType_s3
--- PASS: TestAccVPCFlowLog_LogDestinationType_s3 (21.65s)
=== CONT  TestAccVPCFlowLog_transitGatewayAttachmentID
--- PASS: TestAccVPCFlowLog_transitGatewayID (175.61s)
=== CONT  TestAccVPCFlowLog_LogDestinationType_kinesisFirehose
--- PASS: TestAccVPCFlowLog_transitGatewayAttachmentID (264.53s)
=== CONT  TestAccVPCFlowLog_subnetID
--- PASS: TestAccVPCFlowLog_subnetID (19.00s)
=== CONT  TestAccVPCFlowLog_LogDestinationTypeS3DOParquetHiveCompatible_perHour
--- PASS: TestAccVPCFlowLog_LogDestinationType_kinesisFirehose (146.47s)
=== CONT  TestAccVPCFlowLog_LogDestinationTypeS3DO_parquet
--- PASS: TestAccVPCFlowLog_LogDestinationTypeS3DOParquetHiveCompatible_perHour (18.98s)
=== CONT  TestAccVPCFlowLog_disappears
--- PASS: TestAccVPCFlowLog_LogDestinationTypeS3DO_parquet (18.68s)
=== CONT  TestAccVPCFlowLog_LogDestinationTypeS3DOParquet_hiveCompatible
--- PASS: TestAccVPCFlowLog_disappears (15.63s)
=== CONT  TestAccVPCFlowLog_tags
--- PASS: TestAccVPCFlowLog_LogDestinationTypeS3DOParquet_hiveCompatible (18.87s)
=== CONT  TestAccVPCFlowLog_LogDestinationTypeS3DOPlainText_hiveCompatible
--- PASS: TestAccVPCFlowLog_LogDestinationTypeS3DOPlainText_hiveCompatible (17.75s)
=== CONT  TestAccVPCFlowLog_LogDestinationType_maxAggregationInterval
--- PASS: TestAccVPCFlowLog_tags (40.92s)
=== CONT  TestAccVPCFlowLog_logFormat
--- PASS: TestAccVPCFlowLog_LogDestinationType_maxAggregationInterval (17.21s)
--- PASS: TestAccVPCFlowLog_logFormat (32.05s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	460.028s

```

### CODE REVIEW

File: internal/service/ec2/vpc_flow_log_test.go [hunk 1]
```diff
@@ -227,6 +227,37 @@ func TestAccVPCFlowLog_LogDestinationType_cloudWatchLogs(t *testing.T) {
 	})
 }
 
+func TestAccVPCFlowLog_LogDestinationType_kinesisFirehose(t *testing.T) {
+	var flowLog ec2.FlowLog
+	kinesisFirehoseResourceName := "aws_kinesis_firehose_delivery_stream.test"
+	//s3ResourceName := "aws_s3_bucket.test"
```
Comment (schuylr) [inline internal/service/ec2/vpc_flow_log_test.go:233]: Remove comment

File: internal/service/ec2/vpc_flow_log_test.go [hunk 2]
```diff
@@ -1136,3 +1167,83 @@ resource "aws_flow_log" "test" {
...
+
+resource "aws_flow_log" "test" {
+  log_destination      = aws_kinesis_firehose_delivery_stream.test.arn
+  log_destination_type = "kinesis-data-firehose"
+  traffic_type         = "ALL"
+  vpc_id               = aws_vpc.test.id
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "test" {
+	name        = "kinesis_firehose_test"
```
Comment (schuylr) [inline internal/service/ec2/vpc_flow_log_test.go:1182]: Can you double check that this and below are properly `terraform fmt` formatted?

File: website/docs/r/flow_log.html.markdown [hunk 3]
```diff
@@ -9,7 +9,7 @@ description: |-
 # Resource: aws_flow_log
 
 Provides a VPC/Subnet/ENI/Transit Gateway/Transit Gateway Attachment Flow Log to capture IP traffic for a specific network
-interface, subnet, or VPC. Logs are sent to a CloudWatch Log Group or a S3 Bucket.
+interface, subnet, or VPC. Logs are sent to a CloudWatch Log Group, a S3 Bucket or Amazon Kinesis Data Firehose
```
Comment (schuylr) [inline website/docs/r/flow_log.html.markdown:12]: Suggest that a list of two or more should have an oxford comma at the end :) 

```suggestion
interface, subnet, or VPC. Logs are sent to a CloudWatch Log Group, a S3 Bucket, or Amazon Kinesis Data Firehose
```

**Review by schuylr — APPROVED** [review]
LGTM. I also verified that the `vpc_flow_log.go` file pulls the validation directly from the AWS SDK:

```
ValidateFunc: validation.StringInSlice(ec2.LogDestinationType_Values(), false),
```

Therefore, no modification was necessary for the resource itself. Only acceptance tests and documentation updates were needed.

**Review by schuylr — APPROVED** [review]
LGTM, thanks for implementing those changes and submitting this PR! 🚀

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccVPCFlowLog_LogDestinationType_kinesisFirehose PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccVPCFlowLog_LogDestinationType_kinesisFirehose'  -timeout 180m
=== RUN   TestAccVPCFlowLog_LogDestinationType_kinesisFirehose
=== PAUSE TestAccVPCFlowLog_LogDestinationType_kinesisFirehose
=== CONT  TestAccVPCFlowLog_LogDestinationType_kinesisFirehose
--- PASS: TestAccVPCFlowLog_LogDestinationType_kinesisFirehose (136.62s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	140.774s
```

**Comment by ewbankkit** [comment]
@BhanuCh Thanks for the contribution :tada: :clap:.

**Pull request merged**